### PR TITLE
feat(use-dataloader): handle primitive array key

### PR DESCRIPTION
**Pull request by philibea**

### CODE REVIEW

File: packages/use-dataloader/src/usePaginatedDataLoader.ts [hunk 1]
```diff
@@ -32,9 +32,7 @@ const usePaginatedDataLoader = <ResultType = unknown, ErrorType = Error>(
     perPage = 1,
   }: UsePaginatedDataLoaderConfig<ResultType, ErrorType> = {},
 ): UsePaginatedDataLoaderResult<ResultType, ErrorType> => {
-  if (typeof baseFetchKey !== 'string') {
-    throw new Error(KEY_IS_NOT_STRING_ERROR)
-  }
+  const flatKey = (page: number) => [key, ['page', page]].flat()
```
Comment (chambo-e) [inline packages/use-dataloader/src/usePaginatedDataLoader.ts:35]: need memo here too ?

Comment (philibea) [reply packages/use-dataloader/src/usePaginatedDataLoader.ts:35]: can be usefull right

File: packages/use-dataloader/src/types.ts [hunk 2]
```diff
@@ -1,3 +1,6 @@
+type PrimitiveType = string | number | boolean | null | undefined | Date
```
Comment (QuiiBz) [inline packages/use-dataloader/src/types.ts:1]: I'm not sure why `PrimitiveType` can be `null | undefined`?

File: packages/use-dataloader/src/usePaginatedDataLoader.ts [hunk 3]
```diff
@@ -32,9 +32,10 @@ const usePaginatedDataLoader = <ResultType = unknown, ErrorType = Error>(
     perPage = 1,
   }: UsePaginatedDataLoaderConfig<ResultType, ErrorType> = {},
 ): UsePaginatedDataLoaderResult<ResultType, ErrorType> => {
-  if (typeof baseFetchKey !== 'string') {
-    throw new Error(KEY_IS_NOT_STRING_ERROR)
-  }
+  const flatKey = useMemo(
```
Comment (QuiiBz) [inline packages/use-dataloader/src/usePaginatedDataLoader.ts:35]: Should be a `useCallback` here I think?

**Review by QuiiBz — APPROVED** [review]

File: packages/use-dataloader/src/usePaginatedDataLoader.ts [hunk 4]
```diff
@@ -32,9 +32,10 @@ const usePaginatedDataLoader = <ResultType = unknown, ErrorType = Error>(
     perPage = 1,
   }: UsePaginatedDataLoaderConfig<ResultType, ErrorType> = {},
 ): UsePaginatedDataLoaderResult<ResultType, ErrorType> => {
-  if (typeof baseFetchKey !== 'string') {
-    throw new Error(KEY_IS_NOT_STRING_ERROR)
-  }
+  const flatKey = useCallback(
+    (page: number) => [key, ['page', page]].flat(),
+    [key],
+  )
```
Comment (chambo-e) [inline packages/use-dataloader/src/usePaginatedDataLoader.ts:38]: Nope nope, can we memoize the result of `[key, ['page', page]].flat()` ? :)

File: packages/use-dataloader/src/usePaginatedDataLoader.ts [hunk 5]
```diff
@@ -32,13 +32,11 @@ const usePaginatedDataLoader = <ResultType = unknown, ErrorType = Error>(
...
   }: UsePaginatedDataLoaderConfig<ResultType, ErrorType> = {},
 ): UsePaginatedDataLoaderResult<ResultType, ErrorType> => {
-  if (typeof baseFetchKey !== 'string') {
-    throw new Error(KEY_IS_NOT_STRING_ERROR)
-  }
-
   const [data, setData] = useState<Record<number, ResultType | undefined>>({})
   const [page, setPage] = useState<number>(initialPage ?? 1)
 
+  const keyPage = useMemo(() => [key, ['page', page]].flat(), [key, page])
```
Comment (chambo-e) [inline packages/use-dataloader/src/usePaginatedDataLoader.ts:38]:
```suggestion
  const keyPage = useMemo(() => [...key, 'page', page], [key, page])
```

**Review by chambo-e — APPROVED** [review]

**Pull request merged**